### PR TITLE
Get MoneyMarket balances on selection of Borrower Details

### PR DESCRIPTION
**Pull request by torreyatcitty**
- Refactored Compound balances to track balances by customer addresses so we could support many balances instead of just one.
- Added in translator to Liquidate so Liquidate to trigger the askCustomerBalance msg when an end-user selects a specific Borrow to see more details for.
- UI is placeholder only, will work on better displaying it next.

### CODE REVIEW

File: src/elm/Balances.elm [hunk 1]
```diff
@@ -34,3 +34,45 @@ getBalance balanceDict (Asset assetAddress) ledgerAccount =
...
+        maybeBalance =
+            case maybeAssetDict of
+                Just assetDict ->
+                    maybeMap (Dict.get (ledgerAccountToInt ledgerAccount)) assetDict
+
+                _ ->
+                    Nothing
+    in
+        maybeBalance
+            |> demaybeify
```
Comment (hayesgm) [inline src/elm/Balances.elm]: Off topic, this is actually the "flatten" operation applied to be a maybe. That is: `flattenMaybe :: Maybe Maybe a -> Maybe a` is similar to `flatten List :: List List a -> List a`, etc. The lack of one `flatten` function or one `map` function applied to different types is what gets people uppity about Elm.

File: src/elm/Liquidate.elm [hunk 2]
```diff
@@ -63,13 +67,42 @@ type alias Model =
...
     | SelectBorrow BorrowerDetail
     | Error String
 
 
+type Msg
+    = ForSelf InternalMsg
+    | ForParent OutMsg
+
+
+type alias TranslationDictionary msg =
```
Comment (hayesgm) [inline src/elm/Liquidate.elm]: I've gotten to love Elm's formatting after hating it for a while. It's kind of cool.

File: src/elm/Liquidate.elm [hunk 3]
```diff
@@ -120,8 +153,18 @@ update msg model =
...
 
         SelectBorrow borrowerDetail ->
-            --TODO: Trigger loading of balances (supplies/borrows) here
-            ( { model | selectedBorrow = Just borrowerDetail }, Cmd.none )
+            let
+                loadBalancesCmd =
+                    case contract.moneyMarket of
+                        Just actualMoneyMarket ->
+                            Cmd.map
+                                (ForParent << WrappedCompoundMsg)
```
Comment (hayesgm) [inline src/elm/Liquidate.elm]: Cool function composition!

File: src/elm/Liquidate.elm [hunk 4]
```diff
@@ -270,8 +313,8 @@ oustandingBorrowRow tokens borrowerDetail =
             ]
 
 
-selectedBorrowView : TokenSet -> Maybe BorrowerDetail -> Html Msg
-selectedBorrowView tokens maybeBorrowerDetail =
+selectedBorrowView : CompoundState -> TokenSet -> Maybe BorrowerDetail -> Html Msg
```
Comment (hayesgm) [inline src/elm/Liquidate.elm]: Do we need all of `CompoundState` or can be slim down what we pass? It just means that we need to re-run this function whenever anything in `CompoundState` changes, which could be more expensive if we only depend on a small subset of its state.

File: src/elm/Balances.elm (hunk 1 above)
Comment (torreyatcitty) [reply src/elm/Balances.elm]: oh! ok now that makes sense.

File: src/elm/Liquidate.elm (hunk 4 above)
Comment (torreyatcitty) [reply src/elm/Liquidate.elm]: don't need the whole thing, just the balances. Fixed to be more specific.

**Pull request merged**